code/common/src/test/scala/com/avast/cloud/datadog4s/inmemory/MockMetricsFactory.scala

```diff
@@ -0,0 +1,64 @@
+package com.avast.cloud.datadog4s.inmemory
+
+import java.time.Duration
+
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import com.avast.datadog4s.api.metric._
+import com.avast.datadog4s.api.{ GaugeFactory, HistogramFactory, MetricFactory, Tag }
+
+class MockMetricsFactory[F[_]: Sync](val state: Ref[F, Map[String, Vector[Record[Any]]]]) extends MetricFactory[F] {
```

Comment:
Would it make sense to you to split it for different types: ie. one for timer, one for counter...

Reply:
if the need comes, then yes, but i don't think it adds anything atm

Reply:
i just need something simple to do basic tests, if we decide we need something more sophisticated we can add it

Reply:
Well, it would allow you to get rid of `Any` which would make the use of the mock much easier.